packages/extras/src/lib/components/Environment/Environment.svelte

```diff
@@ -0,0 +1,133 @@
+<script lang="ts">
+	import type { EnvironmentProperties } from '$lib/types/components'
+	import { useThrelte } from '@threlte/core'
+	import { onDestroy } from 'svelte'
+	import {
+		CubeTextureLoader,
+		DefaultLoadingManager,
+		LinearEncoding,
+		PMREMGenerator,
+		sRGBEncoding,
+		Texture,
+		TextureLoader
+	} from 'three'
+	import { HDRCubeTextureLoader } from 'three/examples/jsm/loaders/HDRCubeTextureLoader'
+	import { RGBELoader } from 'three/examples/jsm/loaders/RGBELoader'
+
+	import { GroundProjectedEnv } from 'three/examples/jsm/objects/GroundProjectedEnv'
+
+	export let path: EnvironmentProperties['path'] = undefined
+	export let files: EnvironmentProperties['files']
+	export let isBackground: EnvironmentProperties['isBackground'] = undefined
+	export let groundProjection: EnvironmentProperties['groundProjection'] = undefined
+	export let format: EnvironmentProperties['format'] = 'ldr'
+	export let encoding: EnvironmentProperties['encoding'] = undefined
+
+	const { scene, invalidate, renderer } = useThrelte()
+
+	$: isCubeMap = Array.isArray(files)
+	$: envPath = `${path || ''}${files}`
+
+	let previousEnvPath: string = ``
```

Comment:
Generally when you do these kind of checks, you would immediately assign the current value to a "previous" variable:

```ts
export let someProp: string
let previousProp = someProp
$: if (someProp !== previousProp) {
  // do something
  previousProp = someProp
}
```

This also ensures that the "do something"-part does not run until it actually changes.